include/zephyr/kernel/internal/sched_priq.h

```diff
@@ -44,15 +48,14 @@ void z_priq_rb_remove(struct _priq_rb *pq, struct k_thread *thread);
 struct k_thread *z_priq_rb_best(struct _priq_rb *pq);
 
 /* Traditional/textbook "multi-queue" structure.  Separate lists for a
- * small number (max 32 here) of fixed priorities.  This corresponds
- * to the original Zephyr scheduler.  RAM requirements are
- * comparatively high, but performance is very fast.  Won't work with
- * features like deadline scheduling which need large priority spaces
- * to represent their requirements.
+ * number of fixed priorities.  This corresponds to the original Zephyr
+ * scheduler.  RAM requirements are comparatively high, but performance
+ * is very fast.  Won't work with features like deadline scheduling which
+ * need large priority spaces to represent their requirements.
  */
 struct _priq_mq {
-	sys_dlist_t queues[32];
-	unsigned int bitmask; /* bit 1<<i set if queues[i] is non-empty */
+	sys_dlist_t queues[PRI_QUEUE_BITMAP_SIZE * NBITS];
+	uintptr_t bitmask[PRI_QUEUE_BITMAP_SIZE];
```

Comment:
What do you think of the following @TaiJuWu ?

```cpp
#define NUM_PRIO (CONFIG_NUM_PREEMPT_PRIORITIES + CONFIG_NUM_COOP_PRIORITIES)
#define PRIQ_BITMAP_SIZE (ROUND_UP(NUM_PRIO, sizeof(uintptr_t)) / sizeof(uintptr_t))
```

```suggestion
	sys_dlist_t queues[NUM_PRIO];
	uintptr_t bitmask[PRI_QUEUE_BITMAP_SIZE];
```

Reply:
Great. We can use NHPOT to enforce 2^x if needed.